zcash_client_backend/src/wallet.rs

```diff
@@ -158,111 +223,131 @@ impl transparent_fees::InputView for WalletTransparentOutput {
-/// this output; this will usually be [`zcash_primitives::zip32::Scope`] for received notes, and
-/// `()` for sent notes.
-///
-/// [`OutputDescription`]: sapling::bundle::OutputDescription
-pub struct WalletSaplingOutput<N, S> {
+/// Returns source metadata for a given incoming viewing key.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum KeySource {
+    /// The ZIP 32 account identifier and key scope for the key, if the key was derived from the
+    /// wallet's seed.
+    Seed(AccountId, zip32::Scope),
+    /// The ZIP 32 key scope associated with the key for imported full viewing keys, or `None`
+    /// for imported IVKs.
+    Imported(Option<zip32::Scope>),
+}
+
+impl KeySource {
+    /// Returns the ZIP 32 account identifier if the key was derived from the wallet seed.
+    pub fn account(&self) -> Option<AccountId> {
+        match self {
+            KeySource::Seed(account, _) => Some(*account),
+            KeySource::Imported(_) => None,
+        }
+    }
+}
+
+/// An output that was successfully decrypted in the process of wallet scanning.
+pub struct WalletOutput<Note, Nullifier> {
     index: usize,
-    cmu: sapling::note::ExtractedNoteCommitment,
     ephemeral_key: EphemeralKeyBytes,
-    account: AccountId,
-    note: sapling::Note,
+    note: Note,
     is_change: bool,
     note_commitment_tree_position: Position,
-    nf: N,
-    recipient_key_scope: S,
+    nf: Option<Nullifier>,
+    key_source: KeySource,
 }
 
-impl<N, S> WalletSaplingOutput<N, S> {
-    /// Constructs a new `WalletSaplingOutput` value from its constituent parts.
+impl<Note, Nullifier> WalletOutput<Note, Nullifier> {
+    /// Constructs a new `WalletOutput` value from its constituent parts.
     #[allow(clippy::too_many_arguments)]
     pub fn from_parts(
         index: usize,
-        cmu: sapling::note::ExtractedNoteCommitment,
         ephemeral_key: EphemeralKeyBytes,
-        account: AccountId,
-        note: sapling::Note,
+        note: Note,
         is_change: bool,
         note_commitment_tree_position: Position,
-        nf: N,
-        recipient_key_scope: S,
+        nf: Option<Nullifier>,
+        key_source: KeySource,
     ) -> Self {
         Self {
             index,
-            cmu,
             ephemeral_key,
-            account,
             note,
             is_change,
             note_commitment_tree_position,
             nf,
-            recipient_key_scope,
+            key_source,
         }
     }
 
+    /// The index of the output or action in the transaction that created this output.
     pub fn index(&self) -> usize {
         self.index
     }
-    pub fn cmu(&self) -> &sapling::note::ExtractedNoteCommitment {
-        &self.cmu
-    }
+    /// The [`EphemeralKeyBytes`] value of the transaction output.
     pub fn ephemeral_key(&self) -> &EphemeralKeyBytes {
         &self.ephemeral_key
     }
-    pub fn account(&self) -> AccountId {
```

Comment:
Document this removal (of `WalletSaplingOutput::account`) in the changelog.

Reply:
This is an unnecessary change that will be reverted if I do https://github.com/zcash/librustzcash/pull/1195#discussion_r1505352222; if I don't do that in this PR then it will need to be reintroduced later, which is needless churn.